### PR TITLE
(#788) Add unit tests for Layout.Flat

### PR DESCRIPTION
**Pull request by baudoliver7**
#788
- Unit tests for `Layout.Flat`

### CODE REVIEW

File: src/test/java/com/artipie/LayoutFlatTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * Test case for {@link Layout.Flat}.
+ *
+ * @since 0.15
+ */
+final class LayoutFlatTest {
+
+    @Test
+    void hasNoDashboard() {
+        MatcherAssert.assertThat(
+            "Must not have a dashboard",
```
Comment (genryxy) [inline src/test/java/com/artipie/LayoutFlatTest.java:42]: @baudoliver7 please do not add reason to assertion if  only one assertion in the test method. See this [page](https://github.com/artipie/artipie/blob/master/CONTRIBUTING.md#code-style)

File: src/test/java/com/artipie/LayoutFlatTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link Layout.Flat}.
+ *
+ * @since 0.15
+ */
+final class LayoutFlatTest {
+
```
Comment (genryxy) [inline src/test/java/com/artipie/LayoutFlatTest.java:38]: @baudoliver7 add test case when path consists of one part. And one more test when path is an empty string

**Review by genryxy — CHANGES_REQUESTED** [review]
@baudoliver7 thanks, see comments

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 2 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:38]: @genryxy Ok

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:42]: @genryxy Got it :)

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 2 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:38]: @genryxy Done

**Comment by baudoliver7** [comment]
@genryxy Please, see some new changes.

File: src/test/java/com/artipie/LayoutFlatTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        );
+    }
+
+    @Test
+    void getEmptyKeyFromEmptyPath() {
+        MatcherAssert.assertThat(
+            new Layout.Flat().keyFromPath("").get(),
+            new IsEqual<>(new Key.From(""))
+        );
+    }
```
Comment (genryxy) [inline src/test/java/com/artipie/LayoutFlatTest.java:69]: @baudoliver7 looks like the result of this test is incorrect because for empty string `Optional.empty` should be returned

File: src/test/java/com/artipie/LayoutFlatTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Test
+    void getKeyFromOnePartPath() {
+        MatcherAssert.assertThat(
+            new Layout.Flat().keyFromPath("/one").get(),
+            new IsEqual<>(new Key.From("one"))
+        );
+    }
```
Comment (genryxy) [inline src/test/java/com/artipie/LayoutFlatTest.java:61]: @baudoliver7 please also add test for path with more than one part. It was before but after changes you have deleted this test

**Review by genryxy — CHANGES_REQUESTED** [review]
@baudoliver7 thanks, two comments

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 4 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:61]: @genryxy good

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 3 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:69]: @genryxy I agree with you. To make it work, we should test if key is empty. Let me do that.

File: src/test/java/com/artipie/LayoutFlatTest.java (hunk 4 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/LayoutFlatTest.java:61]: @genryxy Let me introduce a Parameterized Test :)

Comment (genryxy) [reply src/test/java/com/artipie/LayoutFlatTest.java:61]: @baudoliver7 yeah, let's try a parameterized test

**Comment by baudoliver7** [comment]
@genryxy Please, see new changes.

**Review by genryxy — DISMISSED** [review]
@baudoliver7 thanks, looks good to me
@g4s8 please check/merge

**Review by g4s8 — APPROVED** [review]
@baudoliver7 thanks

**Comment by g4s8** [comment]
@baudoliver7 please check: some tests are failing, it's ok with them in master branch

**Comment by baudoliver7** [comment]
@g4s8 I just rebase it with the master branch. All things look fine now :)

**Review by genryxy — APPROVED** [review]
@baudoliver7 thanks

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#854` is not assigned, can't get performer
